### PR TITLE
Xslt transformer

### PR DESCRIPTION
**Pull request by mohdaliiqbal**
This is an XSLT transformer module that uses SI XsltPayloadTransform transformer. It is developed using SI DSL. The project has an integration and module configuration unit tests, as well as one sample xslt file and an xml payload file. Readme has all instructions.

### CODE REVIEW

**Comment by mohdaliiqbal** [comment]
I'm not if this repository or the spring-xd-samples repository is the right place for such contributions. I though this simple module can be useful for someone who wants to ingest xml payloads to any sink after running it through an XSLT. The sample shows how to convert XML to CSV using XSLT. I am not sure if that can be achieved with transformer/splitter oob modules.

**Comment by dturanski** [comment]
I think this is right repo

File: xslt-transformer/src/test/java/org/springframework/xd/xslttransformer/XslTransformerTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+public class XslTransformerTest {
+
+
+	@Test
+	public void test() {
+		AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+		Properties properties = new Properties();
+		properties.put("xslt","classpath:test.xsl");
+		context.getEnvironment().getPropertySources().addLast(new PropertiesPropertySource("options", properties));
+		context.getEnvironment().getPropertySources().addLast(new PropertiesPropertySource("props", properties));
```
Comment (dturanski) [inline xslt-transformer/src/test/java/org/springframework/xd/xslttransformer/XslTransformerTest.java]: This line is a duplicate

File: xslt-transformer/build.gradle [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+If copying this example, it may be simpler to declare the  
+required configuration inline. 
+*/
+buildscript {
+   apply from: 'buildscript.gradle', 
+   to: buildscript
+}
+
+ext {
+    springXdVersion = '1.2.0.M1'
```
Comment (dturanski) [inline xslt-transformer/build.gradle]: Different version than maven

**Comment by dturanski** [comment]
Thank you for the contribution! Many of the modules in the spring-xd-samples repo include both maven and gradle builds. This is mainly to illustrate how modules can be built either way. In practice, it's difficult to maintain both so I would choose one and simplify the README accordingly.  Note that you have different versions of XD declared (1.1.1.RELEASE for maven and 1.2.0.M1 for gradle) here. I am seeing a difference in the contents of the jars and the test fails for the gradle build for me (looking into the cause).  Note that Spring XD 1.2.x includes spring-integration-xml already. I recommend maven since as of 1.2.0.RC1 the parent pom includes dependency management. In cases like this, where you have compile time dependencies on such libraries, you do not need to specify the version.  This raises another point about this repo and which version of XD each module is targeted for. We don't have any policy on that at this point but I will discuss with the team. Meanwhile, I'll look for another commit on this PR to make the recommended changes before merging it. Thanks again!

**Comment by dturanski** [comment]
One more thing - we will need you to sign the [contributor's agreement](https://support.springsource.com/spring_committer_signup), and select the Spring XD project.  This does not grant you commit rights to the repository, but it does mean that we can accept your contributions, and you will get an author credit if we do.  Active contributors might be asked to join the core team, and given the ability to merge pull requests.  Thanks!

**Comment by mohdaliiqbal** [comment]
I have signed the agreement.

The confirmation number is - *124020150601074425*

Thanks,

Muhammad Ali

On Mon, Jun 1, 2015 at 9:47 PM, David Turanski <notifications@github.com>
wrote:

> One more thing - we will need you to sign the contributor's agreement
> <https://support.springsource.com/spring_committer_signup>, and select
> the Spring XD project. This does not grant you commit rights to the
> repository, but it does mean that we can accept your contributions, and you
> will get an author credit if we do. Active contributors might be asked to
> join the core team, and given the ability to merge pull requests. Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spring-projects/spring-xd-modules/pull/13#issuecomment-107652391>
> .
>



-- 
Muhammad Ali
Advisory Consultant

Pivotal Services @ EMC | +971551259736 | mali@pivotal.io
www.pivotal.io

**Comment by dturanski** [comment]
Rebased and merged.

**Pull request closed without merging**